articles/build-a-restful-api-with-adonisjs/index.md

```diff
@@ -0,0 +1,419 @@
+### Build A RESTful API With AdonisJS
+
+### Outline
+Introduction
+Setting up AdonisJS
+Creating Databases
+Setting up Authorization and Authentication
+Creating Models
+Creating Controllers
+Creating Endpoint Routes
+Testing the Forum API
+Conclusion
+
+### Introduction
+In today’s ever-changing and rapid world of frameworks including frontend and backend frameworks, learning and building RESTful APIs to bridge the gap between clients and data is a demanding skill for any software engineer.
+
+"A RESTful API is an application program interface (API) that uses HTTP requests exchange data between parties using endpoints (routes) with either GET, PUT, POST and DELETE as the HTTP requests." from [TowardDataScience](https://towardsdatascience.com)
+
+In this tutorial, we will be showing you how to build a RESTful API with AdonisJS 5. we will be building an API for a forum, we are going to cover a lot such as authentication, authorization, etc. we will also cover best practices in structuring and building out your HTTPS endpoint and also industry standard responses.
+
+AdonisJS is the Laravel of JavaScript and a good grasp of it is a high-demand skill, therefore learning a build your first REST API with it is a game-changer. You can read through the [Ultimate Guide to AdonisJS 5](https://masteringbackend.com/posts/adonisjs-tutorial-the-ultimate-guide) to learn more.
+
+### Setting up AdonisJS 5
+If you’ve been working with JavaScript frameworks before now, you should already have Node.js in your computer, if not, you can install the latest version by going through the steps [here](https://nodejs.org/en/).
+
+_AdonisJS needs Node.js >=12.x.x and NPM >=6.x.x, make sure you have the required NodeJs by checking the version installed in your computer to make sure it corresponds with the requirement._
+
+We will assume you have the requirements, so you can go ahead to create a new AdonisJS 5 project by running this command.
+
+```bash
+    npm init adonis-ts-app adonisjs-forum-api
+```
+
+While you run the command, if you are asked to choose a project structure, select `API Server` and continue by leaving the other options default.
+
+After installing successfully, open the folder with your favorite Text Editor and run the following commands in your terminal.
+
+```bash
+    cd <PROJECT_NAME>
+    node ace serve --watch
+```
+
+Open your browser and visit the URL presented, if you see `hello world`.
+
+Congratulations.
+
+### Creating Databases
+Now that we have our first `Hello World` API created, our Forum API cannot be completed with having a database for storing, reading and updating data from it.
+
+Let’s set up our database right away:
+
+First, you need to create a database with MySQL using any of these [Database Clients](https://masteringbackend.com/posts/top-10-database-clients-for-developers) of your choice.
+
+Next, In other to access and manipulate Databases seemlessly, let's install the LUCID ORM used by AdonisJS with this command.
```

Comment:
`Next, In other to access and manipulate Databases seemlessly` review

Reply:
done